### ISSUE TITLE
Surface should explain its coordinate system

### ISSUE THREAD

**Issue by hcayless** [TEIC/TEI#2148]
[`<surface>`](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-surface.html) does not provide a mechanism for specifying meaning of the numbers in `@ulx`, etc. The [Guidelines](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/PH.html#PHFAX) imply that coordinate system dimensions may be understood to map to units like pixels or millimeters, but lack a way to make this mapping explicit. 

The result is that it's hard to know what the coordinates mean. If `<surface>` had `@width`, `@height`, and either `@units` or the ability to specify units in `@width` and `@height`, then it would be possible (e.g.) to map `<surface>`'s coordinate system to physical dimensions. I suppose an alternative method would be to permit `<surface>` to contain `<dimensions>`.

**Comment by jamescummings** [comment]
What if it had a `@unitRef` pointing to a `<unitDef>`? 

This wouldn't help with the direct mapping but would with the explanation of the units being used.

**Comment by martindholmes** [comment]
I've always assumed that the coordinates are there to _establish_ a matrix, which is independent of the units in which the media file might be delimited ("the surface element defines an abstract coordinate space which may be used to address parts of the image"). So if your image is 200 x 200 px, and you use a 100 by 100 matrix in your `@ulx` etc., then each unit in the surface matrix would correspond to 2px in the image. You can put `@width` and `@height` on the image itself, so the mapping is straightforward. Or am I misunderstanding something?

**Comment by lb42** [comment]
As defined, these numbers are explicitly NOT metricated, as Martin says. They represent a coordinate space, independent of its actual size.  Introducing a unit would really confuse the issue.

**Comment by hcayless** [comment]
Right. What I'd like is to have a way to specify the actual size--that's a separate issue from the definition of the coordinate system (which is correctly abstract).

**Comment by jamescummings** [comment]
Is the correct place for this on zones then?

**Comment by lb42** [comment]
Not on zones either. If you are saying something about the actual size of something in the real world you really don't want to start from here... Try @rend,  last door at the end of the corridor

**Comment by hcayless** [comment]
@jamescummings I don't think so, because if you can define dimensions for a surface then, doing it for zones would be redundant. 

@lb42 I don't think `@rend` would be very useful. I suppose you could do it with `@style`... Hacky though. I kind of like the idea of permitting `<dimensions>` in surface.

**Comment by lb42** [comment]
@hcayless dimensions would be ok. But what if you wanted to define two different surfaces with differing granularity for the same object?

**Comment by hcayless** [comment]
@lb42 would anything stop you from doing that? I'm not sure why you'd want surfaces with different granularity, but you certainly might have surfaces with differing dimensions. Or are you asking whether two surfaces representing the same (e.g.) page could have different coordinate systems? I suppose there'd be no harm, though I don't see an obvious use case for it. You can already do that, of course.

**Comment by lb42** [comment]
I might want one very coarse grained view say 4 X 4 and one very fine day 200 X 100. The dimensions of the objects mapped to those two should be the same, assuming there's only one object.

**Comment by martindholmes** [comment]
The actual dimensions would appear on the media element, wouldn't they? In the simple case, on the `<graphic>`. I'm not seeing why this is actually needed. @hcayless could you outline a real use-case?

**Comment by hcayless** [comment]
Well, the dimensions of the image would (if supplied at all). These are presumably in pixels. What if I want to know how many millimeters the surface is on each side? I guess you're arguing that since a graphic is assumed to be co-extensive with its parent surface or zone, you could infer the container's dimensions from the graphic. And I suppose that would work, though it seems hacky to me--and doesn't help if your surface doesn't have any images.

I feel like there's another issue lurking here with the question of what happens if the graphic's aspect ratio differs from its container's, but maybe let's not go there. I'll mock up an example.

**Comment by martindholmes** [comment]
@hcayless I'm a bit confused as to whether you're trying to describe the original object itself or its digital surrogate. My assumption has always been that the digital surrogate is the subject of the `<surface>` element (i.e. the image representing the object); the object itself would be described in an `<object>` element, presumably in `<physDesc>`.

**Comment by hcayless** [comment]
Assume the papyrus in the image below, with a surface defined as the area outlined in red on the image (i.e. the papyrus itself):

![Screen Shot 2021-05-17 at 3 09 39 PM](https://user-images.githubusercontent.com/23493/118544659-9ebf2f00-b723-11eb-80f9-3315edd4c2ef.jpg)

I want to do:
```xml
<facsimile>
  <surface ulx="152" uly="590" lrx="1060" lry="1378">
    <dimensions>
      <height unit="cm">9.8</width>
      <width unit="cm">11.3</height>
    </dimensions>
    <zone ulx="0" uly="0" lrx="1478" lry="1604">
      <graphic url="https://papyri.info/iiif/6f/AP02272a.jp2/full/3000,/0/default.jpg"/>
    </zone>
  </surface>
</facsimile>
```
I know the dimensions of the papyrus, and I can measure its pixel extent in the image. If I can correlate those, then I can have a viewer that lets (e.g.) users estimate physical letter sizes with a reasonable degree of accuracy. And yes, in this case those dimensions are given in the physDesc, and a method for linking them with the surface might suffice. I don't know that would always be true.

**Comment by martindholmes** [comment]
I have no objection to a dimensions unit being available, but I do think this is a sort of mixing of concerns. The surface element is (to me) an abstract concept wrapping a digital surrogate; it doesn't actually have any dimensions. The digital surrogate does, but of course they're digital. The physical object is a completely separate thing.

But my view of all this may be too simplistic, of course.

**Comment by Martin-de-la-Iglesia** [comment]
If `surface` is only about digital images and not physical objects, why do the Guidelines say that its coordinates can be in millimeters? It makes no sense to me to express coordinates on a digital image in millimeters. Perhaps the Guidelines prose needs to be made clearer there. Then again, it would be highly desirable to use the TEI to define zones on physical objects using metrical units. I don't think this is currently possible in any metadata standard outside of TEI (e.g. SVG, IIIF). Inserting `dimensions` as proposed above seems like a feasible solution to this problem.

**Comment by lb42** [comment]
Is this a a reference to the following passage in the Glines:
       
        It may be most convenient to derive a coordinate space from a digital image of the surface in
         question such that each pixel in the image corresponds with a whole number of units
         (typically 1) in the coordinate space. In other cases it may be more convenient to use
         units such as millimetres. Neither practice implies any specific mapping between the
         coordinate system used and the actual dimensions of the physical object represented.

I think this makes clear that the Glines are only suggesting that a correspondence between the logical dimensions  of the surface and the physical dimensions of the source *may* be convenient. It certaily doesn't say the coordinates are expressed in millimetres. 
Or were you thinking of another passage?

**Comment by Martin-de-la-Iglesia** [comment]
Yes, that is the passage I (and, I think, also Hugh in the original post) was referring to. To me it does indeed say that coordinates can be expressed in pixels, millimeters or other units. If that's not the intended meaning, the passage needs to be rephrased.

**Comment by martindholmes** [comment]
To avoid the interpretation of "expressed in" (which is not what I take from the prose, but still), perhaps we could say "based on" or "derived from", and then add a sentence to the effect that this is an abstract logical matrix which is not expressed in physical units.

**Comment by Martin-de-la-Iglesia** [comment]
OK, then I guess I don't get the meaning of the passage at all. What is the intended relation between millimeters and the numbers in the coordinate attributes? How do you derive an abstract logical matrix from physical units such as millimeters (without simply putting the mm values in the coordinate attibutes)?

**Comment by martindholmes** [comment]
You would use the same numbers as the mm physical dimensions, but the idea is that you're not making any claim as to the correspondence between the two (unless you explicitly say so in your header, I guess). 

If (for example) you have multiple versions of the same image at various resolutions, you can pick a convenient one to form the basis of the surface units, but your offsets can still apply to any of the images at other resolutions because it's a logical matrix rather than a hard physical measurement.

**Comment by Martin-de-la-Iglesia** [comment]
Then that passage is poorly phrased indeed, but isn't the whole point of this thread that some TEI users actually want their coordinate system to be physical rather than abstract?

**Comment by martindholmes** [comment]
As I said before, though, you can establish the correspondence between your matrix and physical units just by adding `@width` and `@height` to the graphic element. You're not prevented from using physical dimensions; you're just not required to map your matrix to physical dimensions if you don't want to (although of course any processor producing usable output from a `<facsimile>` or `<surface>` based on an image would have to do that).

**Comment by Martin-de-la-Iglesia** [comment]
What if there is no `<graphic>` element in my TEI document? Why can't I simply add `@width` and `@height` to `<surface>` (or at least `surface/dimensions`, as proposed above)?

**Comment by martindholmes** [comment]
I'm only responding to your misunderstanding of the intention of the wording in the current prose. I'm not taking any position on the need for a dimensions element.

**Comment by Martin-de-la-Iglesia** [comment]
After discussing this issue at this year's SIG Text & Graphics meeting, I still don't see what harm it could do to introduce something like `@unit` in order to indicate the units of `@ulx` etc. It would be optional, of course - if there's no `@unit` present, the coordinate system is still meant to be abstract.

Having `<dimensions>` within `<surface>` may be a step in the right direction, but it seems a bit semantically weak to me to have something like

```
<surface ulx="0" lrx="123">
   <dimensions unit="mm">
      <width>123</width>
   </dimensions>
</surface>
```

and infer from that that the coordinates are meant to represent millimeters, just because the numbers happen to be the same. Why shouldn't I be able to explicitly state that my coordinates map 1:1 to the mm of the physical object?
